api/vertex.go

```diff
@@ -0,0 +1,75 @@
+package api
+
+import (
+	"github.com/supplyon/gremcos/interfaces"
+)
+
+type vertex struct {
+	builders []interfaces.QueryBuilder
+}
+
+func (v *vertex) String() string {
+
+	queryString := ""
+	for _, queryBuilder := range v.builders {
+		queryString += queryBuilder.String()
+	}
+
+	return queryString
+}
+
+func NewVertex(g interfaces.Graph) interfaces.Vertex {
+	queryBuilders := make([]interfaces.QueryBuilder, 0)
+	queryBuilders = append(queryBuilders, g)
```

Comment:
Does't this mean that could in principle append several graphs to the vertex. I guess this is not what you want. I think we want to associate several vertices with one graph.

Reply:
No the list of query builders is only used to build the query then in the end.
And no it is not possible to add multiple graphs to one vertex if the api (interfaces are used).
Since there is no method allowing this.

Of course inside the package (api) one can do this. But that's not a problem.

Reply:
Agree. Misunderstanding.